http/client/src/main/java/org/wildfly/security/http/client/ElytronHttpClient.java

```diff
@@ -0,0 +1,211 @@
+            digestHttpMechanismUtil.updateAuthParams(authHeader,authParams);
+            request = digestHttpMechanismUtil.createDigestRequest(uri,userName,password,authParams);
+            response = client.send(request, HttpResponse.BodyHandlers.ofString());
+        }
+        return response;
+    }
+
+    //To test header values from ElytronHttpClientTest
+    public HttpRequest getResponseHeader(String responseHeader, String uri) throws Exception {
+
+        digestHttpMechanismUtil.updateAuthParams(responseHeader,authParams);
+
+        String realm = authParams.get("realm");
+        String nonce = authParams.get("nonce");
+        String opaque = authParams.get("opaque");
+        String algorithm = authParams.get("algorithm");
+        String qop = authParams.get("qop");
+
+        String path = digestHttpMechanismUtil.getUriPath(uri);
+
+        userName = httpMechClientConfigUtil.getUsername(new URI(uri));
+        password = httpMechClientConfigUtil.getPassword(new URI(uri));
+
+        String response;
+        if (qop == null) {
+            response = digestHttpMechanismUtil.computeDigestWithoutQop(path, nonce, userName, password, algorithm, realm, "GET");
+        } else {
+            response = digestHttpMechanismUtil.computeDigestWithQop(path, nonce, "0a4f113b", "00000001", userName, password, algorithm, realm, qop, "GET");
+        }
+
+        HttpRequest request2 = HttpRequest
+                .newBuilder()
+                .uri(new URI(uri))
+                .header(AUTHORIZATION, "Digest " +
+                        "username=\"" + userName + "\", " +
+                        "realm=\"" + realm + "\"," +
+                        "nonce=\"" + nonce + "\", " +
+                        "uri=\"" + path + "\", " +
+                        "qop=\"" + qop + "\", " +
+                        "nc=00000001, " +
+                        "cnonce=\"0a4f113b\", " +
+                        "response=\"" + response + "\", " +
+                        "opaque=\"" + opaque + "\", " +
+                        "algorithm=" + algorithm)
+                .build();
+        return request2;
+
+    }
+
+    public HttpRequest getRequest(String uri) throws Exception {
+        String username = httpMechClientConfigUtil.getUsername(new URI(uri));
+        String password = httpMechClientConfigUtil.getPassword(new URI(uri));
+        String AuthType = httpMechClientConfigUtil.getHttpAuthenticationType(new URI(uri));
+        String AuthHeader = null;
+        if (AuthType.equalsIgnoreCase("basic")) {
+            AuthHeader = basicAuth(username, password);
+        }
+        HttpRequest request = HttpRequest
+                .newBuilder()
+                .uri(new URI(uri))
+                .header(AUTHORIZATION, AuthHeader)
+                .build();
+
+        return request;
+    }
+
+    public HttpResponse evaluateBasicMechanism(String uri) throws Exception {
+        HttpRequest request = HttpRequest
```

Comment:
@keshav-725 Just a minor, if we leave each mechanism in this class this class will be too long. Please make a new class for each mechanism.